Project_source/app/src/main/java/android/clicker/school_live_simulator/BicycleState.kt

```diff
@@ -0,0 +1,34 @@
+package android.clicker.school_live_simulator
+
+/**
+ * @author      Andrey Novikov andrewnovicov@gmail.com
+ * @version     1.0                 (current version number of program)
+ * @since       1.0          (the version of the package this class was first added to)
+ */
+abstract class BicycleState {
+    /**
+     * Bike price
+     */
+    protected abstract val price: Int
+
+    /**
+     * A list of available types of delivery.
+     * These types are described in enumeration class Delivery.
+     */
+    protected val available_delivery: ArrayList<Delivery> = arrayListOf<Delivery>()
+
+    /**
+     * Upgrades bicycle.
+     * @param bag player inventory.
+     */
+    abstract fun changeState(bag: Player.Bag)
+
+    /**
+     * Allows delivery.
```

Comment:
I think it is better to write: "Shows that the delivery can be performed"

Reply:
fixed